### ISSUE TITLE
How to update snapshots with unittest?

### ISSUE THREAD

**Issue by codeocelot** [syrusakbary/snapshottest#51]
Appending `--update-snapshot` as is done with nosetest, etc throws `ImportError: No module named --update-snapshot`.  The docs are unclear how exactly to pass this param when using unittest.  Is this features supported when using `unittest`?

**Comment by paulmelnikow** [comment]
Not sure if there is a way to do this, though if there is I'd welcome a PR to add it to the documentation.

**Comment by medmunds** [comment]
I don't believe `--update-snapshot` is implemented when using raw unitttest. Unittest doesn't seem to have a way to add new command-line options for [`unittest.main()`](https://docs.python.org/3.8/library/unittest.html#unittest.main).

You could implement it yourself by setting `snapshot_should_update = True` on the `snapshottest.unittest.TestCase` class when you want to update snapshots. Something like this (warning, untested code):

```python
if __name__ == "__main__":
    import snapshottest.unittest
    import sys
    argv = sys.argv
    if "--update-snapshot" in argv:
        snapshottest.unittest.TestCase.snapshot_should_update = True
        argv = argv.copy()  # (don't modify sys.argv itself)
        argv.remove("--update-snapshot")
    unittest.main(argv=argv)
```

It might make sense to add a new `snapshottest.unitttest.main()` providing this functionality, which people could use in place of `unittest.main()`.

**Comment by paulmelnikow** [comment]
An alternative would be to set it in an environment variable. What do you think about that approach?

**Comment by medmunds** [comment]
I'm of mixed opinions on using an environment variable to request snapshot updates:
* Pro: it would work with `python -m unittest discover`, which is another common way to invoke unittest.
* Con: it's different from all of the other test packages, which support `--update-snapshot` on the command line.
* Con (or Pro?): people would probably expect that environment variable to work with the other test packages then, too.

I think I'd lean toward providing the `unittest.main()` override, to parallel the other test packages. (Switching your unittest's `__main__` is the equivalent of switching your Django `TEST_RUNNER`.) And then as a separate enhancement, *also* adding an environment variable supported for all the test packages. (And working out how that interacts with CI environments—I saw there was some activity around snapshot updates in CI, but haven't looked at the discussion yet.)

Of course, the other way to achieve this is deleting your existing snapshot files and re-running the tests, which will generate new snapshots. Maybe just document that for unittest?

**Comment by medmunds** [comment]
> the other way to achieve this is deleting your existing snapshot files and re-running the tests, which will generate new snapshots

Ah, I see, that's going away in #73. So I guess this does need to get addressed some other way, or it'll be really hard for unittest users to get started with snapshots.

**Comment by davidshepherd7** [comment]
I added an environment variable option to #112 so that it doesn't make things worse for unittest users. 

I think it would be better if someone who uses and understands unittest added a `main` override if we decide we do want one.

**Comment by medmunds** [comment]
I'm much more familiar with Django's unittest wrapper (which has its own test runner and command-line handling), but could take a shot at a `main` override.

Just realized that raw unittest is probably also missing the end-of-test-session handling (delete unvisited snapshots and report summary), which are coded separately for each of the other three test frameworks.
